src/main/kotlin/no/nav/familie/ef/sak/api/beregning/VedtakDto.kt

```diff
@@ -1,53 +1,83 @@
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer
+import com.fasterxml.jackson.module.kotlin.readValue
+import no.nav.familie.ef.sak.api.Feil
+import no.nav.familie.ef.sak.repository.domain.InntektWrapper
+import no.nav.familie.ef.sak.repository.domain.PeriodeWrapper
+import no.nav.familie.ef.sak.repository.domain.Vedtak
+import org.springframework.http.HttpStatus
+import java.util.*
 
 enum class ResultatType {
     INNVILGE,
     AVSLÅ,
     HENLEGGE
 }
+sealed class VedtakDto
+class Henlegge(val resultatType: ResultatType = ResultatType.HENLEGGE) : VedtakDto()
+class Innvilget(val resultatType: ResultatType = ResultatType.INNVILGE,
+                val periodeBegrunnelse: String,
+                val inntektBegrunnelse: String,
+                val perioder: List<VedtaksperiodeDto> = emptyList(),
+                val inntekter: List<Inntekt> = emptyList()) : VedtakDto()
 
-data class VedtakDto(val resultatType: ResultatType,
-                     val periodeBegrunnelse: String,
-                     val inntektBegrunnelse: String,
-                     val perioder: List<VedtaksperiodeDto> = emptyList(),
-                     val inntekter: List<Inntekt> = emptyList())
-
-data class VedtaksperiodeDto(
-        val årMånedFra: YearMonth,
-        val årMånedTil: YearMonth,
-        val aktivitet: String,
-        val periodeType: String
-)
-
-
-fun List<Vedtaksperiode>.fraDomene(): List<VedtaksperiodeDto> =
-        this.map {
-            VedtaksperiodeDto(
-                    årMånedFra = YearMonth.from(it.datoFra),
-                    årMånedTil = YearMonth.from(it.datoTil),
-                    aktivitet = it.aktivitet,
-                    periodeType = it.periodeType,
-            )
-        }
+class Avslå(val resultatType: ResultatType = ResultatType.AVSLÅ,
+            val avslåBegrunnelse: String) : VedtakDto()
+
+fun VedtakDto.tilVedtak(behandlingId: UUID): Vedtak = when (this) {
+    is Avslå -> Vedtak(behandlingId = behandlingId,
+                       avslåBegrunnelse = this.avslåBegrunnelse,
+                       resultatType = ResultatType.AVSLÅ)
+    is Innvilget -> Vedtak(
+            behandlingId = behandlingId,
+            periodeBegrunnelse = this.periodeBegrunnelse,
+            inntektBegrunnelse = this.inntektBegrunnelse,
+            resultatType = ResultatType.INNVILGE,
+            perioder = PeriodeWrapper(perioder = this.perioder.tilDomene()),
+            inntekter = InntektWrapper(inntekter = this.inntekter.tilInntektsperioder()))
+    is Henlegge -> throw Feil("Kan ikke sette vedtak $this då det har feil type", "Kan ikke sette vedtak $this då det har feil type", HttpStatus.BAD_REQUEST)
+}
+
+fun Vedtak.tilVedtakDto(): VedtakDto =
+    when (this.resultatType) {
+        ResultatType.INNVILGE -> Innvilget(
+                resultatType = this.resultatType,
+                periodeBegrunnelse = this.periodeBegrunnelse ?: "",
```

Comment:
Trenger denne å være empty string eller er det greit att den er null?

Reply:
i teorin så borde det vara this.resultatType!! this.periodeBegrunnelse!! ... tror jag?

Reply:
faktiskt inte... frontend har ingen validering på att dom er utfyllda så kan fint vara nullable